app/controllers/user.rb

```diff
@@ -0,0 +1,32 @@
+get '/users' do
+  @user = User.all
+  erb :index
+end
+
+get '/users/new' do
+  @user = User.new
+  erb :'users/new'
+end
+
+get '/users/:id' do
+  @user = User.find_by(id: params[:id])
+  redirect '/' unless @user.id = session[:user_id]
+  erb :'session/login'
+end
+
+post '/users' do
+  user = User.new(params[:user])
+  if user.save
+    session[:user_id] = user.id
+    redirect '/suggestions'
+  else
+    @errors = user.errors.full_messages
+    erb :'/users/new'
+  end
+end
+
+get '/logout' do
+  session[:user_id] = nil
```

Comment:
For security reasons, delete all of the contents in the session hash, even the key/value pairs that Sinatra generates internally:
```ruby
session.clear # 'clear' deletes everything inside the hash, but not the hash itself
```